packages/cli/generation/ir-generator/src/converters/services/convertTransport.ts

```diff
@@ -34,3 +34,59 @@ export async function convertTransport({
         service: protobufService
     });
 }
+
+export async function getTransportForEndpoint({
+    file,
+    serviceTransport,
+    endpointDeclaration,
+    sourceResolver
+}: {
+    file: FernFileContext;
+    serviceTransport: Transport;
+    endpointDeclaration: RawSchemas.HttpEndpointSchema;
+    sourceResolver: SourceResolver;
+}): Promise<Transport | undefined> {
+    const isGrpcService = serviceTransport.type === "grpc";
+    const isHttpService = serviceTransport.type === "http";
+    const isGrpcEndpoint = endpointDeclaration.source !== undefined && "proto" in endpointDeclaration.source;
+    const isHttpEndpoint = !isGrpcEndpoint;
+    if (isHttpService && isHttpEndpoint) {
+        // if the service is http, the endpoint should inherit the service transport
+        return undefined;
+    }
+    if (isHttpService && isGrpcEndpoint) {
+        throw new Error("Cannot have a grpc endpoint on an http service");
+    }
```

Comment:
I imagine we'll eventually want to support this, but it seems fine to keep this as-is for now. This should be impossible for now based on how we determine the service-level transport.

Reply:
My thinking here too!